### PR TITLE
fix(vmm): undo removal of 0xb leaf on AMD for main branch

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

Undo the removal of 0xb leaf on AMD introduced in PR #2164.

## Reason

The removal of 0xb leaf on AMD was introduced in PR #2164, derived from issue #222. As seen in the issue, the reporter told that upgrading guest kernel from 4.19.129 to 4.19.146 fixed the issue, meaning that this was not a KVM (host kernel) side issue.

Actually, leaf 0xb is valid even on AMD, as described in "E.3.8 Function Bh — Extended Topology Enumeration" section of AMD64 Architecture Programmer's Manual Volume 3. And the meaning of each bit is consistent with Intel, as seen in "Table 3-8. Information Returned by CPUID Instruction" section of Intel SDM Vol.2 

AMD64 Architecture Programmer's Manual: https://www.amd.com/system/files/TechDocs/24594.pdf
Intel 64 and IA-32 Architecture Software Developer's Manual: https://www.intel.com/content/www/us/en/developer/articles/technical/intel-sdm.html

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- ~~[ ] This functionality cannot be added in [`rust-vmm`][1].~~

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by zulinx86** [comment]
Should I add a line for this fix into CHANGELOG.md?

File: src/vmm/src/cpuid/transformer/amd.rs [hunk 1]
```diff
@@ -160,22 +165,16 @@ mod tests {
...
-            padding: [0, 0, 0],
-        };
-        cpuid.push(entry).unwrap();
 
         assert!(AmdCpuidTransformer {}
             .process_cpuid(&mut cpuid, &vm_spec)
             .is_ok());
-        assert!(!cpuid.as_slice().contains(&entry));
+
+        // Assert that leaf 0xb / subleaf 1 is generated correctly if not exist in the given cpuid.
```
Comment (kalyazin) [inline src/vmm/src/cpuid/transformer/amd.rs:173]: nit: s/if not exist/if does not exist/

File: src/vmm/src/cpuid/transformer/common.rs [hunk 2]
```diff
@@ -11,6 +11,10 @@ use crate::cpuid::common::get_cpuid;
 // CPUID bits in ebx, ecx, and edx.
 const EBX_CLFLUSH_CACHELINE: u32 = 8; // Flush a cache line size.
 
+// The APIC ID shift in leaf 0xBh specifies the number of bits to shit the x2APIC ID to get a
```
Comment (kalyazin) [inline src/vmm/src/cpuid/transformer/common.rs:14]: nit: s/0xBh/0xb/, s/bits to shit/bits to shift/

**Comment by kalyazin** [comment]
> Should I add a line for this fix into CHANGELOG.md?

I think it is worth doing.

File: src/vmm/src/cpuid/transformer/amd.rs (hunk 1 above)
Comment (zulinx86) [reply src/vmm/src/cpuid/transformer/amd.rs:173]: fixed

File: src/vmm/src/cpuid/transformer/common.rs (hunk 2 above)
Comment (zulinx86) [reply src/vmm/src/cpuid/transformer/common.rs:14]: fixed

**Review by kalyazin — APPROVED** [review]

**Review by JonathanWoollett-Light — APPROVED** [review]

**Pull request merged**